### PR TITLE
refactor auth, rm promises use async js 

### PR DESCRIPTION
**Pull request by kerenketepela**
## Have you read the Contribution Guidelines? If not, check [CONTRIBUTING.md](https://github.com/amos-kibet/mysic-api/blob/main/CONTRIBUTING.md) file first.

(Write your answer here.)

## Description

(Write your description here.)

## Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.

> mark `[X]` to make it checked.

## Related Issues

Fix # (write number after `#`)

### CODE REVIEW

File: config/db.config.js [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 import mongoose from "mongoose";
 import * as dotenv from "dotenv";
 
 dotenv.config();
 const uri = process.env.MONGO_URI;
 const options = {
   dbName: "data",
-  // useNewUrlParser: true,
-  // useUnifiedTopology: true,
+  useNewUrlParser: true,
```
Comment (amos-kibet) [inline config/db.config.js:8]: Any reasons why you have uncommented these? I want to learn 😅

File: controllers/auth.js [hunk 2]
```diff
@@ -1,97 +1,78 @@
...
-        res.json(data);
-      })
-      .catch((err) => {
-        logger.error(err.message);
-        return res.status(500).json({
-          error: `Server error message 2: ${err.message}`,
-        });
-      });
-  }).clone();
+  const checkUser = await User.findOne({ email: email });
```
Comment (amos-kibet) [inline controllers/auth.js:13]: Can we abstract these out of the controller, to services.
I mean, can we abstract out anything related to database to  services, to enforce clean code architecture

**Review by amos-kibet — APPROVED** [review]
lgtm, maybe you can reply my inline comments

File: controllers/auth.js (hunk 2 above)
Comment (kerenketepela) [reply controllers/auth.js:13]: Yeah We can add a services folder and do all querying there

Comment (amos-kibet) [reply controllers/auth.js:13]: Let me work on that

Comment (amos-kibet) [reply controllers/auth.js:13]: You can merge for now, unless you have started working on it

Comment (amos-kibet) [reply controllers/auth.js:13]: You can merge for now, unless you have started working on it

File: config/db.config.js (hunk 1 above)
Comment (kerenketepela) [reply config/db.config.js:8]: To avoid MongoDB warnings as below version 5.7

Comment (amos-kibet) [reply config/db.config.js:8]: Some of these options bring warnings when used😅😅. I have never understood which ones to use and which ones not to use.

**Pull request merged**